src/main/kotlin/app/CodeLongevity.kt

```diff
@@ -0,0 +1,173 @@
+        revWalk.markStart(head);
+
+        var childCommit = head;
+        for (commit in revWalk) {
+            //println("\nparent $commit")
+            //println("this $childCommit")
+
+            val df = DiffFormatter(DisabledOutputStream.INSTANCE)
+            df.setRepository(repo)
+            df.setDetectRenames(true)
+            val diffs = df.scan(commit, childCommit);
+            //println("filesChanged: ${diffs.size}")
+
+            for (diff in diffs) {
+                // todo: skip binary files.
+
+                // A new file, no deleted lines.
+                if (diff.changeType == DiffEntry.ChangeType.ADD) {
+                    continue
+                }
+
+                val file = diff.getOldPath()
+                //println("old file: $file, new file: ${diff.getNewPath()}")
+
+                var blame: BlameResult? = null
+                //var lines: RawText? = null
+                var delta = 0
+
+                for (edit in df.toFileHeader(diff).toEditList()) {
+                    val delStart = edit.getBeginA()
+                    val delEnd = edit.getEndA()
+                    val delCount = edit.getLengthA()
+                    val insCount = edit.getLengthB()
+
+                    // Delete case.
+                    if (delCount > 0) {
+                        if (blame == null) {
+                            var blameGen = BlameGenerator(repo, file)
+                            blameGen.push(null, commit)
+                            blame = blameGen.computeBlameResult()
+                            // If no blame, skip it. It may happen, for example,
+                            // when a change is an empty folder removal.
+                            if (blame == null) {
+                                break
+                            }
+                            //lines = blame.getResultContents()
+                        }
+
+                        var index = delStart + delta
+                        //println("original file index: $index")
+
+                        while (index < delEnd) {
+                            val blameCommit = blame.getSourceCommit(index)
+                            //val deletedLine = lines!!.getString(index)
+                            //val blameDate = Date(blameCommit.getCommitTime().toLong() * 1000)
+                            val timeDiff = (childCommit.getCommitTime() - blameCommit.getCommitTime())
+                            val author = blame.getSourceAuthor(index)
+                            val authorName = author.getName()
+                            val authorEmail = author.getEmailAddress()
+
+                            list.add(CodeLine(blameCommit.getName(),
+                                     file, index,
+                                     authorName, authorEmail, timeDiff))
+                            index++
+                        }
+                    }
+
+                    delta += insCount - delCount
```

Comment:
It is possible due to different number of lines added and deleted in previous partition edition of current file. JGit dealing with offsets instead of you. So you have Edit per partition edition of file. Begin/end a and begin/end b is absolute index of line for old and new file respectively.